### ISSUE TITLE
Focus is not working

### ISSUE THREAD

**Issue by BackOrder** [Shoes3/shoes3#32]
The **focus method** does not work unless called using a timer (*see workaround*). The elements **edit_box** and **list_box** will not work at all (*see limitations*).

```ruby
Shoes.app {
   e = edit_line
   e.focus
}
```

**TEMPORARY WORKAROUND**
```ruby
Shoes.app {
   e = edit_line

   timer(0) { e.focus }
}
```

**LIMITATIONS**
The following code demonstrates that even with the workaround **edit_box** and **list_box** do not focus. It iterates through the array of elements that support focus method and invoke focus on each of them by an increment of time.

```ruby
Shoes.app {
   e = []
   e << button("button")
   e << edit_box
   e << edit_line
   e << radio
   e << check
   e << (list_box :items => [1, 2, 3])

   (0..e.size - 1).each { |n|
      timer(n * 5) {
         info "#{n * 5} seconds: #{e[n].class}"
         e[n].focus
      }
   }
}
```

**Comment by BackOrder** [comment]
The feature implemented in issue #34 demonstrated that the scope of elements may not be available at the time they are called due to how Shoes process script initialization. The following script cause an error whereas it should not:

```ruby
Shoes.app {
   @c = check
   para @c.checked?
}
```

> (cshoes.exe:8752): Gtk-CRITICAL **: gtk_toggle_button_get_active: assertion `GTK_IS_TOGGLE_BUTTON (toggle_button)' failed

Using a timer will allow the block to be executed only once elements are initialized and prevent the Gtk-CRITICAL error.

```ruby
Shoes.app {
   @c = check
   timer(0) { para @c.checked? }
}
```

**Comment by ccoupe** [comment]
One can also use the start method instead of a timer.  I'm going to close this issues since there is nothing anybody is going to do. It's just part of Shoes.

**Comment by BackOrder** [comment]
It doesn't make much sense to close it for the aforementioned reason. There is nothing wrong to let it open until some solutions come up. It's not a closing issues contest! It could however be label low priority if you feel the challenge versus reward is too big.

**Comment by ccoupe** [comment]
You're alive @BackOrder ?  I cruise the issues to determine what to fix next and I come across issues that no one is going to fix because thats 'just shoes' or no one is going fix without a complete top-down re-think/re-write and I don't any volunteers for that effort - they're probably busy writing cross platform app GUI apps in javascript.  If you want to assign yourself to this issue, then please reopen it.
